drivers/flash/flash_stm32.c

```diff
@@ -120,7 +120,7 @@ int flash_stm32_wait_flash_idle(struct device *dev)
 		return -EIO;
 	}
 
-	while ((FLASH_STM32_REGS(dev)->sr & FLASH_SR_BSY)) {
+	while ((FLASH_STM32_REGS(dev)->SR & FLASH_SR_BSY)) {
```

Comment:
For some reason, on G0 this is not defined and FLASH_SR_BSY1 should be used instead.
You could also use FLASH_FLAG_BSY but this adds a dependency on HAL.
The best alternative might be to handle this variation here locally and maybe raise an issue directly on STM32Cube package here: https://github.com/STMicroelectronics/STM32CubeG0

Reply:
I see. If that's the case, then I will put ifdef guards around that line so that it is different just for the G0 till it gets updated in STM32Cube. (Didn't know STM32Cube was open source)